Content.Server/Interaction/InteractionSystem.cs

```diff
@@ -471,47 +469,49 @@ public void DoAttack(EntityUid user, EntityCoordinates coordinates, bool wideAtt
+            if (TryComp(user, out HandsComponent? hands))
             {
-                if (hands.GetActiveHand?.Owner is {Valid: true} item)
+                var item = hands.GetActiveHand?.Owner;
+
+                if (item != null && !Deleted(item.Value))
                 {
                     if (wideAttack)
                     {
-                        var ev = new WideAttackEvent(item, user, coordinates);
-                        RaiseLocalEvent(item, ev, false);
+                        var ev = new WideAttackEvent(item.Value, user, coordinates);
+                        RaiseLocalEvent(item.Value, ev, false);
 
                         if (ev.Handled)
                         {
-                            _adminLogSystem.Add(LogType.AttackArmedWide, LogImpact.Medium, $"{ToPrettyString(user):user} wide attacked with {ToPrettyString(item):used} at {coordinates}");
+                            _adminLogSystem.Add(LogType.AttackArmedWide, LogImpact.Medium, $"{ToPrettyString(user):user} wide attacked with {ToPrettyString(item.Value):used} at {coordinates}");
                             return;
                         }
                     }
                     else
                     {
-                        var ev = new ClickAttackEvent(item, user, coordinates, targetUid);
-                        RaiseLocalEvent(item, ev, false);
+                        var ev = new ClickAttackEvent(item.Value, user, coordinates, targetUid);
+                        RaiseLocalEvent(item.Value, ev, false);
 
                         if (ev.Handled)
                         {
-                            if (targetUid != default)
+                            if (targetUid != null)
                             {
                                 _adminLogSystem.Add(LogType.AttackArmedClick, LogImpact.Medium,
-                                    $"{ToPrettyString(user):user} attacked {ToPrettyString(targetUid):target} with {ToPrettyString(item):used} at {coordinates}");
+                                    $"{ToPrettyString(user):user} attacked {ToPrettyString(targetUid.Value):target} with {ToPrettyString(item.Value):used} at {coordinates}");
                             }
                             else
                             {
                                 _adminLogSystem.Add(LogType.AttackArmedClick, LogImpact.Medium,
-                                    $"{ToPrettyString(user):user} attacked with {ToPrettyString(item):used} at {coordinates}");
+                                    $"{ToPrettyString(user):user} attacked with {ToPrettyString(item.Value):used} at {coordinates}");
                             }
 
                             return;
                         }
                     }
                 }
-                else if (!wideAttack && targetUid != default && _entityManager.HasComponent<ItemComponent>(targetUid))
+                else if (!wideAttack && targetUid != null && HasComp<ItemComponent>(targetUid.Value))
```

Comment:
And here?

Reply:
`HasComp` already checks for deleted, pretty much. It will return false if the entity doesn't exist or is deleted